modules/javafx.base/src/test/java/test/com/sun/javafx/collections/ObservableMapWrapperTest.java

```diff
@@ -0,0 +1,191 @@
+
+            var map2 = new ObservableMapWrapper<>(new HashMap<>(Map.of("k0", "v0", "k1", "v1", "k2", "v2")));
+            assertThrows(NullPointerException.class, () -> map2.entrySet().removeAll((Collection<?>) null));
+        }
+
+        @Test
+        public void testKeySetNullArgumentThrowsNPE() {
+            var map1 = new ObservableMapWrapper<>(new HashMap<>());
+            assertThrows(NullPointerException.class, () -> map1.keySet().removeAll((Collection<?>) null));
+
+            var map2 = new ObservableMapWrapper<>(new HashMap<>(Map.of("k0", "v0", "k1", "v1", "k2", "v2")));
+            assertThrows(NullPointerException.class, () -> map2.keySet().removeAll((Collection<?>) null));
+        }
+
+        @Test
+        public void testValueSetNullArgumentThrowsNPE() {
+            var map1 = new ObservableMapWrapper<>(new HashMap<>());
+            assertThrows(NullPointerException.class, () -> map1.values().removeAll((Collection<?>) null));
+
+            var map2 = new ObservableMapWrapper<>(new HashMap<>(Map.of("k0", "v0", "k1", "v1", "k2", "v2")));
+            assertThrows(NullPointerException.class, () -> map2.values().removeAll((Collection<?>) null));
+        }
+
+        @Test
+        public void testRemoveAllKeysWithEmptyCollectionArgumentDoesNotEnumerateBackingMap() {
+            ObservableMapWrapper<String, String> map = newNonIterableObservableMapWrapper();
+            map.keySet().removeAll(Collections.<String>emptySet());
+        }
+
+        @Test
+        public void testRemoveAllValuesWithEmptyCollectionArgumentDoesNotEnumerateBackingMap() {
+            ObservableMapWrapper<String, String> map = newNonIterableObservableMapWrapper();
+            map.values().removeAll(Collections.<String>emptySet());
+        }
+
+        @Test
+        public void testRemoveAllEntriesWithEmptyCollectionArgumentDoesNotEnumerateBackingMap() {
+            ObservableMapWrapper<String, String> map = newNonIterableObservableMapWrapper();
+            map.entrySet().removeAll(Collections.<Map.Entry<String, String>>emptySet());
+        }
+
+        @Test
+        public void testRemoveAllEntriesWithEmptyCollectionArgumentWorksCorrectly() {
+            var content = Map.of("k0", "v0", "k1", "v1", "k2", "v2");
+            var map = new ObservableMapWrapper<>(new HashMap<>(content));
+            assertFalse(map.entrySet().removeAll(Set.of()));
+            assertEquals(content, map);
+        }
+    }
+
+    @Nested
+    class RetainAllTest {
+        @Test
+        public void testEntrySetNullArgumentThrowsNPE() {
+            var map1 = new ObservableMapWrapper<>(new HashMap<>());
+            assertThrows(NullPointerException.class, () -> map1.entrySet().retainAll((Collection<?>) null));
+
+            var map2 = new ObservableMapWrapper<>(new HashMap<>(Map.of("k0", "v0", "k1", "v1", "k2", "v2")));
+            assertThrows(NullPointerException.class, () -> map2.entrySet().retainAll((Collection<?>) null));
```

Comment:
Would advise to rename `map1` to `emptyMap` and `map2` to `notEmptyMap` or the like.

Same in other places.

Reply:
Done.